docs/developer-guides/Scripting_Terraform.md

```diff
@@ -0,0 +1,176 @@
+- **Modularisation and reusability**: Terraform facilitates the packaging of infrastructure into modular components, enhancing both reusability and ease of sharing across organisational teams.
+- **State management**: Terraform's state management capabilities ensure an accurate representation of real-world resources, enabling features such as resource dependencies and idempotency.
+- **Collaboration and workflow**: The platform supports collaboration through features like remote backends and state locking, thereby fostering collective work on infrastructure projects.
+- **Community and ecosystem**: A robust community actively contributes to the Terraform ecosystem, providing a wealth of modules and examples that expedite infrastructure development.
+
+## Features
+
+Here are some key features built into this repository's Terraform module:
+
+- Provides Make targets for frequently-used Terraform commands for streamlined execution
+- Offers code completion and command signature assistance via Make for enhanced CLI usability
+- Supports named arguments with default values for an improved coding experience
+- Allows the working directory to be controlled by either arguments or a predefined constant for flexible stack management
+- Features a command wrapper to improve the onboarding experience and ensure environmental consistency
+- Incorporates both a Git hook and a GitHub action to enforce code quality standards
+- Comes with the CI/CD pipeline workflow integration
+- Includes a file cleanup routine to efficiently remove temporary resources
+- Incorporates a ready-to-run example to demonstrate the module's capabilities
+- Integrates a code linting routine to ensure scripts are free from unintended side effects
+- Includes a verbose mode for in-depth troubleshooting and debugging
+- Incorporates a best practice guide
+
+## Key files
+
+- Scripts
+  - [terraform.lib.sh](../../scripts/terraform/terraform.lib.sh) A library code loaded by custom make targets and CLI scripts
+  - [terraform.mk](../../scripts/terraform/terraform.mk): Customised implementation of the Terraform routines loaded by the `scripts/init.mk` file
+  - [terraform.sh](../../scripts/terraform/terraform.sh): Terraform command wrapper
+- Configuration
+  - [.tool-versions](../../.tool-versions): Stores Terraform version to be used
+- Code quality gates
+  - [lint-terraform/action.yaml](../../.github/actions/lint-terraform/action.yaml): GitHub action
+  - [check-terraform-format.sh](../../scripts/githooks/check-terraform-format.sh): Git hook
+- Usage example
+  - Declarative infrastructure definition example [terraform-state-aws-s3](../../scripts/terraform/examples/terraform-state-aws-s3) to store Terraform state
+  - A set of [make targets](https://github.com/nhs-england-tools/repository-template/blob/main/scripts/terraform/terraform.mk#L44) to run the example
+
+## Usage
+
+### Quick start
+
+Run the example:
+
+TODO: vscode integration ?
```

Comment:
Thought that the AWS Toolkit for Visual Studio code might have avoided the need for exporting the keys and tokens but can't see any documentation suggesting it works in the terminal window.

Reply:
I have searched internally for any recommendations and found that both `awsume` and `aws-vault` are commonly used. This makes me think that we should not include any opinion on that in the repository template (for now), as it belongs to the `dotfiles` project where we already have the following:

- aws and azure [tooling](https://github.com/nhs-england-tools/dotfiles/blob/74808560933c4f12dd0ab330a7bef692d509370b/assets/30-install-developer-tools.macos.sh#L63) being installed
- vscode aws [extension](https://github.com/nhs-england-tools/dotfiles/blob/74808560933c4f12dd0ab330a7bef692d509370b/.vscode/extensions.json#L4) on the list of recommended ones